app/services/cabinet-mail.js

```diff
@@ -55,7 +56,7 @@ export default class CabinetMailService extends Service {
       };
 
       const mail = caseSendBackEmail(params);
-      const creator = await submission.creator;
+      const creator = await this.draftSubmissionService.getCreator(submission);
       const mailResource = this.store.createRecord('email', {
```

Comment:
I noticed that these mail methods are being called without being awaited. If the user quits to soon the mails might not have been created